code/game/gamemodes/wizard/artefact.dm

```diff
@@ -597,3 +597,231 @@ var/global/list/multiverse = list()
+
+
+/obj/item/device/necromantic_stone/nya
+	name = "nya-cromantic stone"
+	desc = "A shard capable of resurrecting humans as creatures of Vile Heresy. Even the Wizard Federation fears it.."
+	icon = 'icons/obj/wizard.dmi'
+	icon_state = "nyacrostone"
+	item_state = "electronic"
+	origin_tech = "bluespace=4;materials=4"
+	w_class = 1
+	heresy = 1
+	unlimited = 1
+
+/////////////////////////////////////////Voodoo///////////////////
+
+
+/obj/item/voodoo
+	name = "wicker doll"
+	desc = "Something creepy about it."
+	icon = 'icons/obj/wizard.dmi'
+	icon_state = "voodoo"
+	item_state = "electronic"
+	var/mob/living/carbon/human/target = null
+	var/list/mob/living/carbon/human/possible = list()
+	var/obj/item/link = null
+	var/cooldown_time = 30 //3s
+	var/cooldown = 0
+
+
+
+/obj/item/voodoo/attackby(obj/item/I as obj, mob/user as mob, params)
+	if(target && cooldown < world.time)
+		if(is_hot(I))
+			target << "<span class='userdanger'>You suddenly feel very hot</span>"
+			target.bodytemperature += 50
+			GiveHint(target)
+		else if(can_puncture(I))
+			target << "<span class='userdanger'>You feel a stabbing pain in [parse_zone(user.zone_sel.selecting)]!</span>"
+			target.Weaken(2)
+			GiveHint(target)
+		else if(istype(I,/obj/item/weapon/bikehorn))
+			target << "<span class='userdanger'>HONK</span>"
+			target << 'sound/items/AirHorn.ogg'
+			target.ear_damage += rand(0,3)
+			GiveHint(target)
+		cooldown = world.time +cooldown_time
+		return
+	if(!link)
+		if(I.loc == user && istype(I) && I.w_class <= 2)
+			user.drop_item()
+			I.loc = src
+			link = I
+			user << "You attach [I] to the doll."
+			update_targets()
+	..()
+/obj/item/voodoo/check_eye(mob/user as mob)
+	return src.loc == user
+
+/obj/item/voodoo/attack_self(mob/user as mob)
+	if(!target && possible.len)
+		target = input(user, "Select your victim!", "Voodoo") as null|anything in possible
+		return
+	if(user.zone_sel.selecting == "chest")
+		if(link)
+			target = null
+			link.loc = get_turf(src)
+			user << "<span class='notice'>You remove the [link] from the doll.</span>"
+			link = null
+			update_targets()
+			return
+	if(target && cooldown < world.time)
+		switch(user.zone_sel.selecting)
+			if("mouth")
+				var/wgw =  sanitize(input(user, "What would you like the victim to say", "Voodoo", null)  as text)
+				target.say(wgw)
+				log_game("[user][user.key] made [target][target.key] say [wgw] with a voodoo doll.")
+			if("eyes")
+				user.set_machine(src)
+				if(user.client)
+					user.client.eye = target
+					user.client.perspective = EYE_PERSPECTIVE
+				spawn(100)
+					user.reset_view()
+					user.unset_machine()
+			if("r_leg","l_leg")
+				user << "<span class='notice'>You move the doll's legs around.</span>"
+				var/turf/T = get_step(target,pick(cardinal))
+				target.Move(T)
+			if("r_arm","l_arm")
+				//use active hand on random nearby mob
+				var/list/nearby_mobs = list()
+				for(var/mob/living/L in range(target,1))
```

Comment:
```range(target,1)``` is technically wrong too!, I once spent an hour debugging because I did ```orange(thing,5)``` instead of ```orange(5,thing)```...